File: python/cookbook/util/logging.py
#!/usr/bin/env python
# -*- coding:utf-8 -*-

import logging

class LoggerFactory(object):

    __loggers = dict()

    @staticmethod
    def getLogger(name):

        if name in LoggerFactory.__loggers:
            return LoggerFactory.__loggers[name]

        handler = logging.StreamHandler()
        handler.setFormatter(logging.Formatter(
            "%(asctime)s [#%(process)-5s - %(threadName)-8s] %(module)s: %(levelname)-5s %(message)s"
        ))

        logger = logging.getLogger(name)
        logger.setLevel(logging.DEBUG)
        logger.addHandler(handler)

        LoggerFactory.__loggers[name] = logger

        return logger

File: ruby/cookbook/util/logging.rb
#!/usr/bin/env ruby

module LoggerFactory

    require 'logger'

    class << self

        @@__loggers = Hash.new

        def getLogger(name)

            if @@__loggers.key?(name) then
                return @@__loggers[name]
            end

            logger = Logger.new(STDOUT)
            logger.level = Logger::DEBUG
            logger.formatter = proc { |severity, datetime, progname, msg|
                "#{datetime.localtime('+09:00').iso8601} [##{$$.to_s.ljust(5)} - #{Thread.current[:name].ljust(8)}] #{name}: #{severity.ljust(5)} #{msg} \n"
            }

            @@__loggers[name] = logger

            return logger

        end

    end

end
